### PR TITLE
[Enhancement] Keep the number of migration tasks running on be to a fixed number

### PR DESCRIPTION
**Pull request by gengjun-git**
Currently, if there are huge number of tablets to migrate (SSD->HDD or HDD->SSD), FE will send 1000 tasks to BE every 1 minute(tablet report interval), which will cause tasks to pile up on BE, and end up with OOM.
We can use AgentTaskQueue to get the number of task running on BE, and send (Config.tablet_sched_max_migration_task_sent_once - running_tasks) tasks to BE, so that the number of migration tasks running on BE can be kept in a fixed number. 

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

**Review by trueeyu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java [hunk 1]
```diff
@@ -304,6 +298,50 @@ public void tabletReport(long backendId, Map<Long, TTablet> backendTablets,
...
+        if (backendStorageTypeCnt <= 1) {
+            LOG.debug("available storage medium type count is less than 1, " +
+                            "no need to send migrate task. tabletId={}, backendId={}.",
+                    tabletMeta, backendId);
+            return;
+        }
+
+        // 2. If size of tabletMigrationMap exceeds (Config.tablet_sched_max_migration_task_sent_once - running_tasks_on_be),
+        // dot not send more tasks. The number of tasks running on be cannot exceed Config.tablet_sched_max_migration_task_sent_once
+        if (tabletMigrationMap.size() >=
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TabletInvertedIndex.java:315]: the tasks in tabletMigrationMap is sent in ReportHandler.handleMigration

File: fe/fe-core/src/main/java/com/starrocks/leader/LeaderImpl.java [hunk 2]
```diff
@@ -722,43 +723,47 @@ private void finishClone(AgentTask task, TFinishTaskRequest request) {
...
+                // local migration just set path hash
+                Replica replica = GlobalStateMgr.getCurrentInvertedIndex().getReplica(tabletId, task.getBackendId());
+                Preconditions.checkArgument(reportedTablet.isSetPath_hash());
+                replica.setPathHash(reportedTablet.getPath_hash());
+            } finally {
+                db.writeUnlock();
+            }
         } finally {
-            db.writeUnlock();
+            AgentTaskQueue.removeTask(task.getBackendId(), TTaskType.STORAGE_MEDIUM_MIGRATE, task.getSignature());
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/leader/LeaderImpl.java:766]: The migration task is not added into AgentTaskQueue before. I added the tasks into AgentTaskQueue at ReportHandler.handleMigration

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 46 (13.04%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|25|00.00%|[288, 729, 730, 731, 732, 736, 737, 738, 741, 742, 743, 744, 745, 746, 749, 750, 751, 752, 753, 756, 759, 760, 761, 763, 766]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|5|20|25.00%|[315, 317, 318, 322, 323, 327, 328, 329, 331, 333, 334, 336, 339, 342, 343]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|1|1|100.00%|[]|

**Review by nshangyiming — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 51 (13.73%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|25|00.00%|[288, 729, 730, 731, 732, 736, 737, 738, 741, 742, 743, 744, 745, 746, 749, 750, 751, 752, 753, 756, 759, 760, 761, 763, 766]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|5|24|20.83%|[315, 317, 318, 319, 320, 324, 325, 326, 330, 331, 332, 334, 336, 337, 338, 340, 343, 346, 347]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/task/AgentTaskQueue.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 52 (15.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|0|25|00.00%|[288, 729, 730, 731, 732, 736, 737, 738, 741, 742, 743, 744, 745, 746, 749, 750, 751, 752, 753, 756, 759, 760, 761, 763, 766]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|5|24|20.83%|[315, 317, 318, 319, 320, 324, 325, 326, 330, 331, 332, 334, 336, 337, 338, 340, 343, 346, 347]|
|:large_blue_circle:|com/starrocks/common/proc/LocalTabletsProcDir.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/task/AgentTaskQueue.java|1|1|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

**Review by trueeyu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 9 / 53 (16.98%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/leader/LeaderImpl.java|1|26|03.85%|[286, 727, 728, 729, 730, 734, 735, 736, 739, 740, 741, 742, 743, 744, 747, 748, 749, 750, 751, 754, 757, 758, 759, 761, 764]|
|:large_blue_circle:|com/starrocks/catalog/TabletInvertedIndex.java|5|24|20.83%|[315, 317, 318, 319, 320, 324, 325, 326, 330, 331, 332, 334, 336, 337, 338, 340, 343, 346, 347]|
|:large_blue_circle:|com/starrocks/common/proc/LocalTabletsProcDir.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/leader/ReportHandler.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/task/AgentTaskQueue.java|1|1|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**